### ISSUE TITLE
"Typical setup" mode of failure is unacceptably confusing

### ISSUE THREAD

**Issue by mike-north** [typed-ember/ember-cli-typescript#360]
Because of the strong conventions in Ember's ecosystem, and due to the precedent set by v1 of this library, consumers assume that getting the v2 working with their app is approximately the following steps

1) 
```sh
ember install ember-cli-typescript
```

2) Start writing typescript

--- 

~#359~ #362  takes care of clarifying that step 1 should instead be
```sh
ember install ember-cli-typescript --save
```
for ember addons (not apps) to ensure it's installed as a `dependency`, but we should also validate that the environment/installation is reasonable in our `included()` hook.

The current mode of failure is one or more "loader cannot find module XYZ" errors, and without deep familiarity with this project, consumers are likely to chase down many unrelated paths before finally considering the `devDependency` vs. `dependency` issue.

The established behavior of [`ember-cli-babel`](https://github.com/ember-cli/ember-cli/blob/6a4c515ff62e6a0a5d1c2055ad393171062ffd5e/lib/models/addon.js#L1329-L1338) or [`ember-cli-htmlbars`](https://github.com/ember-cli/ember-cli/blob/6a4c515ff62e6a0a5d1c2055ad393171062ffd5e/lib/models/addon.js#L1134-L1137) would be a fine solution, although in this case, the check should be in this library and not ember-cli.

**Comment by chriskrycho** [comment]
I’ll see if I can hit this tomorrow or Monday, unless someone beats me to it.

**Comment by chriskrycho** [comment]
Fixed in fff704eb4cd37668fbbda993ba283d47d14a66b5.
